packages/react/src/components/Button/Button.tsx

```diff
@@ -144,21 +154,28 @@ class Button extends UIComponent<WithAsProp<ButtonProps>, ButtonState> {
         {Box.create(!hasChildren && content, {
           defaultProps: { as: 'span', styles: styles.content },
         })}
-        {!hasChildren && iconPosition === 'after' && this.renderIcon(variables, styles)}
+        {!hasChildren && iconPosition === 'after' && !loading && this.renderIcon(variables, styles)}
       </ElementType>
     )
   }
 
   renderIcon = (variables, styles) => {
-    const { icon, iconPosition, content } = this.props
-
-    return Icon.create(icon, {
-      defaultProps: {
-        styles: styles.icon,
-        xSpacing: !content ? 'none' : iconPosition === 'after' ? 'before' : 'after',
-        variables: variables.icon,
-      },
-    })
+    const { icon, iconPosition, content, loading, loader } = this.props
+
+    return loading
+      ? Loader.create(loader, {
+          defaultProps: {
+            size: 'smallest',
+            styles: styles.loader,
+          },
+        })
+      : Icon.create(icon, {
+          defaultProps: {
+            styles: styles.icon,
+            xSpacing: !content ? 'none' : iconPosition === 'after' ? 'before' : 'after',
+            variables: variables.icon,
+          },
+        })
```

Comment:
```jsx
const loaderElement = loading && Loader.create()
const iconElement = !loading && Icon.create()
```

If we will specify this inline in `renderComponent()`, will it be more clear?

Reply:
Actually we are rendering both elements now, and the theme can hide one of them if necessary.